packages/p2p/src/components/my-ads/my-ads.jsx

```diff
@@ -14,63 +16,56 @@ const MyAdsState = ({ message }) => (
     </div>
 );
 
-class MyAds extends React.Component {
-    state = {
-        show_form: false,
-    };
-
-    handleShowForm = show_form => {
-        this.setState({ show_form });
-    };
-
-    componentDidMount() {
-        this.is_mounted = true;
+const MyAds = () => {
+    const { is_advertiser, is_restricted, setPoiStatus } = React.useContext(Dp2pContext);
+    const [is_loading, setIsLoading] = React.useState(true);
+    const isMounted = useIsMounted();
+    const [show_ad_form, setShowAdForm] = React.useState(false);
 
-        if (!this.context.is_advertiser) {
-            requestWS({ get_account_status: 1 }).then(response => {
-                if (this.is_mounted && !response.error) {
-                    const { get_account_status } = response;
-                    const { authentication } = get_account_status;
-                    const { identity } = authentication;
-                    const { status } = identity;
-
-                    this.setState({
-                        poi_status: status,
-                    });
-                }
-
-                this.setState({ is_loading: false });
-            });
-        } else {
-            this.setState({ is_loading: false });
+    React.useEffect(() => {
+        if (isMounted()) {
+            if (!is_advertiser) {
+                requestWS({ get_account_status: 1 }).then(response => {
+                    if (isMounted() && !response.error) {
+                        const { get_account_status } = response;
+                        const { status } = get_account_status.authentication.identity;
+                        setPoiStatus(status);
+                    }
+                });
+            }
+            setIsLoading(false);
```

Comment:
`is_loading` is set to `false` immediately after the component being mounted. This state update should happen inside the the `then` scope.